File: app/src/main/java/ar/com/wolox/android/example/ui/home/HomeModule.java
package ar.com.wolox.android.example.ui.home;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class HomeModule {

    @ContributesAndroidInjector
    protected abstract HomeActivity homeActivity();

    @ContributesAndroidInjector
    protected abstract HomeFragment homeFragment();

}

File: app/src/main/java/ar/com/wolox/android/example/ui/home/HomeModule.kt
package ar.com.wolox.android.example.ui.home

import ar.com.wolox.android.example.ui.news.NewsFragment
import ar.com.wolox.android.example.ui.profile.ProfileFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class HomeModule {

    @ContributesAndroidInjector
    protected abstract fun homeActivity(): HomeActivity

    @ContributesAndroidInjector
    protected abstract fun homeFragment(): HomeFragment

    @ContributesAndroidInjector
    protected abstract fun newsFragment(): NewsFragment

    @ContributesAndroidInjector
    protected abstract fun profileFragment(): ProfileFragment
}
